ci/jenkins/stop-stale-jobs.sh

```diff
@@ -0,0 +1,166 @@
+    
+        capv_job_name=$(echo "$capv_jobs" | jq -r "[.[]][10][$i].name")
+
+        # Job color is the status of the jobs, for example if jobs are in running state then its value will be {"red_anime","blue_anime","aborted_anime".....}
+        # so here "*_anime" means job is in running state.
+        # red_anime: last failed job is in running state.
+        # blue_anime: last successful job is in running state.
+        # aborted_anime: last aborted job is in running state.   
+        job_color=$(echo "$capv_jobs" | jq -r "[.[]][10][$i].color")
+
+        if [[ ${job_color: -5} == "anime" ]]; then
+            last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq -r "[.[]][10][0].number")
+            jenkins_variable="$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json")"
+            pr_id=$(echo "$jenkins_variable" | jq -r "[.[]][1].ghprbPullId")
+
+            if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+                echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+                # Need to stop first then clean.
+                curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN || true
+                
+                # Ensures that the job has been aborted before cleanup.
+                retry=5
+                flag=false
+                while [[ "${retry}" -gt 0 ]]; do
+                    status=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/$last_build_number/api/json" | jq -r "[.[]][14]")
+                    if [[ "$status" == "ABORTED" ]]; then
+                         flag=true
+                         break
+                    fi
+                    sleep 5
+                    retry=$((retry-1)) 
+                done
+                if [[ $flag == true ]]; then
+
+                    CLUSTER_NAME=$(echo "$jenkins_variable" | jq -r "[.[]][1].BUILD_TAG")
+                    ip_of_capv_testbed=$(echo "$jenkins_variable" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+
+                    cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+                else
+                    echo "Job $capv_job_name still running on PR ${PULL_REQUEST_ID}"
+                fi
+            fi
+        fi   
+    done
+}
+
+function stop_all() {
+     # abort all stale waiting jobs on PR.
+      abort_queue_jobs
+
+     # abort all stale running jobs on PR.
+      abort_running_jobs   
+}
+
+function release_static_ip {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo "=== Releasing IP ==="
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} \
+    "cat \"$DEFAULT_WORKDIR/host-local.json\" | \
+    CNI_COMMAND=DEL CNI_CONTAINERID=\"$CLUSTER_NAME\" \
+    CNI_NETNS=/dev/null CNI_IFNAME=dummy0 CNI_PATH=. /usr/bin/host-local"
+    echo "Released IP"
+}
+
+function cleanup_running_job {
```

Comment:
Get exit status in variable for cluster delete and cluster namespace delete case and handle error
if [[ $kubectl_delete_cluster_exit_code -eq 0 && $kubectl_delete_ns_exit_code -eq 0 ]]; then
        echo "=== Cluster ${CLUSTER_NAME} cleanup succeeded ==="
    else
        echo "=== Cluster ${CLUSTER_NAME} cleanup failed ==="
        # Additional error handling or cleanup actions can be added here
    fi